x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -301,13 +301,13 @@ export const SearchBar: FC<SearchBarProps> = ({
           setShowAppend(false);
         },
         onBlur: () => {
-          setShowAppend(true);
+          setShowAppend(!searchValue.length);
         },
         fullWidth: true,
         append: showAppend ? (
           <EuiFormLabel
             title={keyboardShortcutTooltip}
-            css={{ fontFamily: euiTheme.font.familyCode }}
+            css={{ fontFamily: euiTheme.font.familyCode, position: 'absolute', right: 0 }}
```

Comment:
Why is the absolute positioning necessary?

Reply:
So that when removing / adding back the _append_ component it does not affect the parent container's size.
This way we completely avoid a small layout shift that is visually disturbing.

Reply:
This is because the width is being applied at the search input instead of the overall wrapping element. I'll describe the fix a review comment.

Reply:
Then you can delete this absolute positioning.
```suggestion
            css={{ fontFamily: euiTheme.font.familyCode }}
```